Helper/ManagementHelper.php

```diff
@@ -63,19 +63,34 @@ public function __construct(StoreManager $storeManager, Data $adyenHelper, Confi
 
     /**
      * @param string $xapikey
+     * @param bool|null $demoMode
      * @return array
      * @throws AdyenException
      * @throws NoSuchEntityException
      */
-    public function getMerchantAccountWithApikey(string $xapikey, ?bool $demoMode = null): array
+    public function getMerchantAccountAndClientKey(string $xapikey, ?bool $demoMode = null): array
     {
         $storeId = $this->storeManager->getStore()->getId();
         $client = $this->adyenHelper->initializeAdyenClient($storeId, $xapikey, $demoMode);
         $management = new Management($client);
-        $response = $management->me->retrieve();
+        $responseMe = $management->me->retrieve();
+        $associatedMerchantAccounts = [];
+        $page = 1;
+        $pageSize = 100;
+        //get the associated merchant accounts using get /merchants.
+        $responseMerchants = $management->merchantAccount->list(["pageSize" => $pageSize]);
+        foreach ($responseMerchants['data'] as $value) {
+            array_push($associatedMerchantAccounts, $value['id']);
+        }
+        while (isset($responseMerchants['_links']['next'])) {
+            $responseMerchants = $management->merchantAccount->list(["pageSize" => $pageSize, "pageNumber" => ++$page]);
+            foreach ($responseMerchants['data'] as $value) {
+                array_push($associatedMerchantAccounts, $value['id']);
+            }
```

Comment:
Then here we merge other pages:
```suggestion
            $associatedMerchantAccounts += array_column($responseMerchants['data'], 'id');
```